shap/plots/__init__.py

```diff
@@ -19,6 +21,9 @@
 from ._violin import violin
 from ._waterfall import waterfall
 
+plt.rcParams['figure.constrained_layout.use'] = True
+
+
```

Comment:
I'm not so fully convinced that this is a good idea. 1) In my experience constrained_layout has caused some other issues (I can't quite communicate what off the top of my head, but there are some trade-offs for sure). 2) Global rcparams like these might interfere with the user's own config (?), I'm not really in favour of this.

I noted that you mentioned about some issues with ylabels being cut-off from a change in dataset. That's a little unusual, do you think it's worth to raise an issue reproducing the problem (without setting constrained layout here) and let's investigate/brainstorm if there are alternative ways of solving this?

Reply:
See #3213. I'll leave this in draft until this issue is resolved.